frontend/src/metabase/admin/settings/components/Email/SettingsEmailForm.jsx

```diff
@@ -0,0 +1,61 @@
+import { Component } from "react";
+import { push } from "react-router-redux";
+import PropTypes from "prop-types";
+import { connect } from "react-redux";
+
+import { getSetting } from "metabase/selectors/settings";
+import { getIsEmailConfigured } from "metabase/setup/selectors";
+import { SettingsSection } from "../../app/components/SettingsEditor/SettingsSection";
+import { SMTPConnectionCard } from "./SMTPConnectionCard";
+
+class SettingsEmailForm extends Component {
```

Comment:
Since you've made significant changes to this file, wdyt about converting it to TS and a functional style?